topics/S3/etcd-migration/src/steps.py

```diff
@@ -0,0 +1,196 @@
+              f"script (^C) fix what is needed and re-run script{bcolors.ENDC}")
+        wait_for_drain(host)
+        print(f"{bcolors.DARK_GREEN}\tHost {host} is in Drain Mode, Restarting it into new mode{bcolors.ENDC}")
+
+        output = send_bash_command(f"weka debug manhole --slot 0 --host {int(''.join(filter(str.isdigit, host)))} "
+                                   f"s3_update_config force_minio_refresh=true")
+        while True:
+            time.sleep(2)
+            output = send_bash_command("weka s3 cluster status -J")
+            hosts_health = json.loads(output)
+            if all(val for val in hosts_health.values()):
+                break
+
+    print(f"{bcolors.GREEN}\tAll hosts are Ready.{bcolors.ENDC}")
+
+
+@step("Validate all minIOs are up with KWAS")
+def validate_kwas_is_up():
+    for host in global_vars.s3_hosts:
+        r = send_request(f"http://{host}:9001/minio/migration/clients")
+        response_body = r.json()
+        if not (response_body['kwas client'] == "true" and response_body['etcd client'] == "false"):
+            print(f"{bcolors.RED}ERROR: Not all minIOs are up with kwas. Fix it and run script again. The invalid data is: {bcolors.ENDC}")
+            exit(1)
+    print(f"{bcolors.GREEN}\tAll hosts are Ready with KWAS.{bcolors.ENDC}")
+
+    print(f"{bcolors.YELLOW}{bcolors.BOLD}After this step, rolling back to ETCD is no longer supported. "
+          f"Meaning this is the step after which the S3 cluster is forever KWAS.{bcolors.ENDC}")
+    global_vars.auto_mode = 1
+
+
+@step("Exit S3 cluster from Migration Read Only Mode")
+def exit_migration_mode():
+    for host in global_vars.s3_hosts:
+        output = send_bash_command(f"weka debug manhole --slot 0 --host {int(''.join(filter(str.isdigit, host)))} s3_exit_kwas_migration_mode")
+    for host in global_vars.s3_hosts:
+        r = send_request(f"https://{host}:9000/minio/migration/ro-mode")
+        response_body = r.json()
+        if response_body["Mode"] != "false":
+            print(f"{bcolors.RED}ERROR: Host {host} failed to exit Migration Read Only Mode. Fix what is needed and rerun script{bcolors.ENDC}")
+            exit(1)
+
+    print(f"{bcolors.GREEN}\tAll S3 hosts exited Migration Read Only Mode.{bcolors.ENDC}")
+
+
+@step("Stop ETCD from running, and Change minIO defaults to KWAS")
+def remove_etcd_internals():
+    output = send_bash_command(f"weka s3 cluster update --etcd-enable=off")
+    for host in global_vars.s3_hosts:
+        output = send_bash_command(f"ssh {host} weka local exec -C s3 supervisorctl stop etcd")
+
+    print(f"{bcolors.GREEN}\tKWAS set to default, current etcd clients stopped.{bcolors.ENDC}")
+
+
+def do_revert():
+    global_vars.step_number = 1
+    global_vars.auto_mode = 2
+    print(f"{bcolors.YELLOW}{bcolors.BOLD}Starting Revert{bcolors.ENDC}")
+
+    enter_migration_mode(True)
+    drain_and_restarts_all_hosts()
+    print(f"{bcolors.GREEN}Revert finished successfully{bcolors.ENDC}")
```

Comment:
maybe add post revert validations. for example that on all minios etcd client is used instead of kwas client